### PR TITLE
proposal: Segment API Redesign

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: rfcs/0-segment.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+type Segmenter interface {
+    ListSegments(pairs ...Pair) SegmentIterator
+    InitSegment(path string, pairs ...Pair) Segment
+    AbortSegment(seg Segment) error
+}
+
+type IndexSegmenter interface {
+    Segmentable
```
Comment (Prnyself) [inline rfcs/0-segment.md:82]: `Segmentable` means `Segmenter`?

Comment (Xuanwo) [reply rfcs/0-segment.md]: It's a typo, fixed.

File: rfcs/0-segment.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+author: Xuanwo <github@xuanwo.io>
+status: draft
+updated_at: 2019-11-15
```
Comment (Prnyself) [inline rfcs/0-segment.md:4]: fix update time plz

**Review by Prnyself — CHANGES_REQUESTED** [review]

File: rfcs/0-segment.md (hunk 2 above)
Comment (Xuanwo) [reply rfcs/0-segment.md]: fixed

**Review by Prnyself — APPROVED** [review]

**Pull request merged**